enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,995 @@
+
+## Summary
+
+The goal of this enhancement is to provide the ability to install in a
+vSphere environment with multiple datacenters and clusters.
+
+This will be an opinionated design, the vCenter datacenter will always be a `region`
+and a vCenter cluster will always be a `zone`.
+
+## Motivation
+
+Users of OpenShift would like the ability to deploy
+within multiple datacenters and clusters to increase
+reliability. Customers would also like to take advantage
+of the concept of regions and zoning that this type of deployment would
+offer.
+
+- https://issues.redhat.com/browse/RFE-845
+- https://issues.redhat.com/browse/OCPPLAN-4927
+
+### Goals
+
+- Support regions and zones in vSphere using multiple datacenters (region) and
+clusters (zone)
+- Support installation into multiple datacenters and multiple clusters
+
+### Non-Goals
+
+- Support multiple subnets
+- Support multiple vcenters
+
+Note: The platform spec will be modified to support this at a future date.
+Only a single item in `vcenters` will be supported for the initial release.
+
+## Proposal
+
+Modification of the installer to support the provisioning of masters in defined
+datacenters and clusters.
+
+### api
+
+#### Infrastructure
+
+Using the upstream [vSphere cluster api](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/apis/v1beta1/vspherefailuredomain_types.go)
+and the upstream [vSphere cloud config manager](https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/common/config/types_yaml.go)
+as examples to implement parameters of `VSpherePlatformSpec`.
+These parameters include the optional and required information to manage a OpenShift cluster on vSphere.
+
+```golang
+// VSpherePlatformFailureDomainType is the name of the failure domain type.
+// There are two defined failure domains currently, Datacenter and ComputeCluster.
+// Each represents a vCenter object type within a vSphere environment.
+// +kubebuilder:validation:Enum="";HostGroup;Datacenter;ComputeCluster
+type VSpherePlatformFailureDomainType string
+
+const (
+    // HostGroup as a type allows the use of a group of ESXi hosts
+    // to be represented as a failure domain zone. When using this
+    // case it is expected that region would be a cluster.
+    // HostGroups within vCenter must be preconfigured and
+    // assigned in the topology.
+    HostGroupFailureDomain      VSpherePlatformFailureDomainType = "HostGroup"
+
+    // ComputeCluster failure domain can either be a zone or region.
+    // The vCenter cluster is required to preconfigured and
+    // assigned in the topology.
+    ComputeClusterFailureDomain VSpherePlatformFailureDomainType = "ComputeCluster"
+
+    // Datacenter failure domain can be only a region. The vcenter
+    // datacenter is required to be preconfigred and assigned
+    // in the topology. If used the zone would be of type ComputeCluster.
+    DatacenterFailureDomain     VSpherePlatformFailureDomainType = "Datacenter"
```

Comment:
I'm not sure if it is better to leave these comments here, or in openshift/api#1199, but:

I think this wording is based on upstream. We should probably update it to affect the truth for OpenShift as proposed here. Particularly:

- HostGroupFailureDomain -> not supported
- ComputeClusterFailureDomain -> can only be a zone

Reply:
>     * HostGroupFailureDomain -> not supported

Oh, I see this now in the API PR in the comment for `VSpherePlatformFailureDomainType`: ` // + HostGroup is included for future support, consumers should output an error`